server/reframe/reframe.go

```diff
@@ -0,0 +1,70 @@
+package reframe
+
+import (
+	"bytes"
+	"net/http"
+
+	"github.com/filecoin-project/go-indexer-core"
+	"github.com/filecoin-project/storetheindex/internal/registry"
+	"github.com/filecoin-project/storetheindex/server/finder/handler"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-delegated-routing/client"
+	"github.com/ipfs/go-delegated-routing/server"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/routing"
+	"github.com/multiformats/go-multicodec"
+	"github.com/multiformats/go-multihash"
+	"github.com/multiformats/go-varint"
+)
+
+func NewReframeHTTPHandler(indexer indexer.Interface, registry *registry.Registry) http.HandlerFunc {
+	return server.DelegatedRoutingAsyncHandler(NewReframeService(handler.NewFinderHandler(indexer, registry)))
+}
+
+func NewReframeService(fh *handler.FinderHandler) *ReframeService {
+	return &ReframeService{finderHandler: fh}
+}
+
+type ReframeService struct {
+	finderHandler *handler.FinderHandler
+}
+
+func (x *ReframeService) FindProviders(key cid.Cid) (<-chan client.FindProvidersAsyncResult, error) {
+	mh := key.Hash()
+	fr, err := x.finderHandler.Find([]multihash.Multihash{mh})
+	if err != nil {
+		return nil, err
+	}
+	ch := make(chan client.FindProvidersAsyncResult)
```

Comment:
Will there always be a reader for this channel?  If not, then make it buffered so that the writer goroutine does not hang/leak when there is no reader.

Also, if there are cases when there may not be a reader, then consider having the caller pass in a channel if they want to reveive an async result.